services/outputhost/outputhost.go

```diff
@@ -507,22 +507,16 @@ func (h *OutputHost) ReceiveMessageBatch(ctx thrift.Context, request *cherami.Re
 	moreResultTimer := common.NewTimer(timeoutTime.Sub(time.Now()))
 	defer moreResultTimer.Stop()
 MORERESULTLOOP:
-	for remaining := count - 1; remaining > 0; {
+	for remaining := count - 1; remaining > 0; remaining-- {
 		select {
-		case msg := <-cgCache.msgsRedeliveryCh:
-			deliver(msg, cgCache.msgCacheRedeliveredCh)
-			remaining--
-		case msg := <-cgCache.msgsCh:
-			deliver(msg, cgCache.msgCacheCh)
-			remaining--
+		case <-moreResultTimer.C:
+			break MORERESULTLOOP
 		default:
```

Comment:
then i don't think the "default" makes sense here.. let's get rid of that one as well.